src/utils/tabs.rs

```diff
@@ -0,0 +1,74 @@
+use crate::config::SPACES;
+
+use unicode_segmentation::UnicodeSegmentation;
+
+pub fn clamp_arg(tab_size: usize) -> usize {
+    std::cmp::min(tab_size, SPACES.len())
+}
+
+/// Expand tabs as spaces.
+/// tab_width = 0 is documented to mean do not replace tabs.
+pub fn expand(line: &str, tab_width: usize) -> String {
+    let line_bytes = line.as_bytes();
+    if tab_width == 0 || !line_bytes.iter().any(|c| *c == b'\t') {
+        line.to_string()
+    } else {
+        let expanded = Vec::with_capacity(line.len());
+
+        let expand_tabs = |mut v: Vec<u8>, c: &u8| -> Vec<u8> {
+            if *c == b'\t' {
+                v.extend_from_slice(SPACES[..tab_width].as_bytes());
+            } else {
+                v.push(*c);
+            }
+
+            v
+        };
+        let expanded = line_bytes.iter().fold(expanded, expand_tabs);
```

Comment:
Do you see an argument for the `fold` over

```rust
        for c in line_bytes.iter() {
            if *c == b'\t' {
                expanded.extend_from_slice(SPACES[..tab_width].as_bytes());
            } else {
                expanded.push(*c);
            }
        }
```

here?

Reply:
Less `mut` and thus more idiomatic? And maybe the `fold` one vectorizes better (well, I checked on godbolt, neither do currently...)